libnetwork/datastore/datastore.go

```diff
@@ -219,36 +197,37 @@ func NewDataStoreFromConfig(dsc discoverapi.DatastoreConfigData) (DataStore, err
 		return nil, fmt.Errorf("cannot parse store configuration: %v", dsc.Config)
 	}
 
-	scopeCfg := ScopeCfg{
+	ds, err := New(ScopeCfg{
 		Client: ScopeClientCfg{
 			Address:  dsc.Address,
 			Provider: dsc.Provider,
 			Config:   sCfgP,
 		},
-	}
-
-	ds, err := NewDataStore(scopeCfg)
+	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to construct datastore client from datastore configuration %v: %v", dsc, err)
 	}
 
 	return ds, err
 }
 
-func (ds *datastore) Close() {
+// Close closes the data store.
+func (ds *Store) Close() {
 	ds.store.Close()
 }
 
-func (ds *datastore) Scope() string {
+// Scope returns the scope of the store.
+func (ds *Store) Scope() string {
 	return ds.scope
 }
 
-func (ds *datastore) KVStore() store.Store {
+// KVStore returns access to the KV Store.
+func (ds *Store) KVStore() store.Store {
```

Comment:
This method is only used by a few tests. We can probably get rid of it.

Reply:
Oh, good one; can probably take care of that in https://github.com/moby/moby/pull/46041, which is a follow-up to this one 👍